roles/kubernetes/preinstall/templates/resolved.conf.j2

```diff
@@ -1,6 +1,10 @@
 [Resolve]
+{% if dns_early is sameas true and dns_late is sameas false %}
```

Comment:
Looks like this bug doesn't only affect resolvd.conf mode (`0061-systemd-resolved.yml)`. It also appears to affect plain resolv.conf mode too (`0060-resolvconf.yml`), and maybe even the others.

Maybe we should fix the bug at a higher level? Like maybe adding a `when: not dns_early` conditional to all of the dns-altering tasks in [main.yaml](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/kubernetes/preinstall/tasks/main.yml)?

Reply:
I can only fix something that had appeared and tested in my environment. Whether this also exists as a problem in the other dns configuration options, I therefore do not know. I would also assume that in early mode you don't need to change the dns configuration, as you should be able to assume an existing and working dns configuration. So your suggestion would be better. But, i think, there was a reason why the early/late mode including the configuration change was introduced. But I don't know that reason, so I just added a small fix for systemd-resolved.

Reply:
Looks like the real culprit is this PR: https://github.com/kubernetes-sigs/kubespray/pull/7731

That landed in v2.18, which revealed that the early/late DNS distinction was never working as intended. I can confirm that this bug does also affect `0060-resolvconf.yml`.

I can help provide an all-encompassing fix for #8560 . Will do some testing and let you know.

Reply:
yes, I also think so. The missing fix (like appending the if statement with early/late check) for [roles/kubernetes/preinstall/tasks/0040-set_facts.yml](https://github.com/kubernetes-sigs/kubespray/pull/7731/files#diff-5ec85c416b4149ff215b482e62a12173e3f1bd184f0b3501440087ca43caf014) should help with resolvconf.

I think resolvconf and systemd-resolved are too different to find an all-encompassing fix.